src/provider/default.ts

```diff
@@ -172,10 +166,9 @@ export class Provider implements ProviderInterface {
   public async getStorageAt(
     contractAddress: string,
     key: number,
-    blockHash?: BigNumberish,
-    blockNumber: BlockNumber = null
+    blockIdentifier: BlockIdentifier = null
   ): Promise<object> {
-    const formattedBlockIdentifier = getFormattedBlockIdentifier(blockHash, blockNumber);
+    const formattedBlockIdentifier = getFormattedBlockIdentifier(blockIdentifier);
```

Comment:
Would it be easier to pass in `BlockIdentifierObject` instead of `blockIdentifier` here?